Tests/BugsnagKVStoreTest.m

```diff
@@ -0,0 +1,167 @@
+//
+//  BugsnagKVStoreTest.m
+//  Bugsnag-iOSTests
+//
+//  Created by Karl Stenerud on 11.09.20.
+//  Copyright © 2020 Bugsnag Inc. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "BugsnagKVStore.h"
+#import "BSGCachesDirectory.h"
+ 
+@interface BugsnagKVStoreTest : XCTestCase
+
+@end
+
+@implementation BugsnagKVStoreTest
+
+- (void)setUp {
+    int err = 0;
+    NSString* path = [[BSGCachesDirectory cachesDirectory] stringByAppendingPathComponent:@"bsgkv"];
+    bsgkv_open([path cStringUsingEncoding:NSUTF8StringEncoding], &err);
+    XCTAssertEqual(err, 0);
+}
+
+- (void)tearDown {
+    bsgkv_close();
+}
+
+#define DECLARE_SCALAR_TEST(FUNC, TYPE, EXPECTED1, EXPECTED2) \
+- (void)test##FUNC { \
+    const char* name = "test-" #TYPE; \
+    TYPE expected = EXPECTED1; \
+    TYPE expected2 = EXPECTED2; \
+    TYPE actual = 0; \
+    int err = 0; \
+\
+    bsgkv_delete(name, &err); \
+    XCTAssertEqual(err, 0); \
+\
+    bsgkv_delete(name, &err); \
+    XCTAssertEqual(err, 0); \
+\
+    bsgkv_set##FUNC(name, expected, &err); \
+    XCTAssertEqual(err, 0); \
+\
+    actual = bsgkv_get##FUNC(name, &err); \
+    XCTAssertEqual(err, 0); \
+    XCTAssertEqual(actual, expected); \
+\
+    bsgkv_set##FUNC(name, expected2, &err); \
+    XCTAssertEqual(err, 0); \
+\
+    actual = bsgkv_get##FUNC(name, &err); \
+    XCTAssertEqual(err, 0); \
+    XCTAssertEqual(actual, expected2); \
+\
+    bsgkv_delete(name, &err); \
+    XCTAssertEqual(err, 0); \
+\
+    actual = bsgkv_get##FUNC(name, &err); \
+    XCTAssertEqual(err, ENOENT); \
+\
+    bsgkv_set##FUNC(name, expected, &err); \
+    XCTAssertEqual(err, 0); \
+\
+    actual = bsgkv_get##FUNC(name, &err); \
+    XCTAssertEqual(err, 0); \
+    XCTAssertEqual(actual, expected); \
+}
+
+DECLARE_SCALAR_TEST(Boolean, bool, true, false)
+DECLARE_SCALAR_TEST(Int, int64_t, 1000, 2000)
+DECLARE_SCALAR_TEST(Float, double, 123.456, 7.89e50)
+
+
+- (void)testString {
+    const char* name = "test-string";
+    const char* expected = "Blah blah blah blah blah";
+    const char* expected2 = "Something else";
+    char actual[100];
+    int err = 0;
+    
+    bsgkv_delete(name, &err);
+    XCTAssertEqual(err, 0);
+    
+    bsgkv_delete(name, &err);
+    XCTAssertEqual(err, 0);
+
+    bsgkv_setString(name, expected, &err);
+    XCTAssertEqual(err, 0);
+
+    bsgkv_getString(name, actual, sizeof(actual), &err);
+    XCTAssertEqual(err, 0);
+    XCTAssertEqual(strcmp(actual, expected), 0);
```

Comment:
It would be helpful for future travelers (and potential CI breakage) to print something more descriptive about the failure, like the expected and actual values, particularly since `strcmp()` will fall over in the event of `actual` being NULL.

Maybe an assertion wrapper? Something like:

```c
#define AssertEqualChars(actual, expected) \
    actual == NULL \
    ? XCTFail("Expected '%s', got NULL", expected) \
    : XCTAssertEqual(strcmp(actual, expected), 0, "Expected '%s', got '%s'", expected, actual)
```